code/modules/spacepods/spacepod.dm

```diff
@@ -362,75 +399,83 @@
+			remove_equipment(user, equipment_system.cargo_system, "cargo_system")
+			return
+		if("Secondary Cargo System")
+			remove_equipment(user, equipment_system.sec_cargo_system, "sec_cargo_system")
+			return
 		if("Lock System")
-			SPE = equipment_system.lock_system
-			if(user.put_in_any_hand_if_possible(SPE))
-				to_chat(user, "<span class='notice'>You remove \the [SPE] from the equipment system.</span>")
-				SPE.removed(user)
-				SPE.my_atom = null
-				equipment_system.lock_system = null
-			else
-				to_chat(user, "<span class='warning'>You need an open hand to do that.</span>")
-		/*
-		if("engine system")
-			SPE = equipment_system.engine_system
-			if(user.put_in_any_hand_if_possible(SPE))
-				to_chat(user, "<span class='notice'>You remove \the [SPE] from the equipment system.</span>")
-				equipment_system.engine_system = null
-			else
-				to_chat(user, "<span class='warning'>You need an open hand to do that.</span>")
-		if("shield system")
-			SPE = equipment_system.shield_system
-			if(user.put_in_any_hand_if_possible(SPE))
-				to_chat(user, "<span class='notice'>You remove \the [SPE] from the equipment system.</span>")
-				equipment_system.shield_system = null
-			else
-				to_chat(user, "<span class='warning'>You need an open hand to do that.</span>")
-		*/
+			remove_equipment(user, equipment_system.lock_system, "lock_system")
+			return
+	return
+
+/obj/spacepod/proc/remove_equipment(mob/user, var/obj/item/device/spacepod_equipment/SPE, var/slot)
 
+	if(passengers.len > max_passengers - SPE.occupant_mod)
+		to_chat(user, "<span class='warning'>Someone is sitting in \the [SPE]!</span>")
+		return
+
+	var/sum_w_class = 0
+	for(var/obj/item/I in cargo_hold.contents)
+		sum_w_class += I.w_class
+	if(cargo_hold.contents.len > cargo_hold.storage_slots - SPE.storage_mod["slots"] || sum_w_class > cargo_hold.max_combined_w_class - SPE.storage_mod["w_class"])
+		to_chat(user, "<span class='warning'>Empty \the [SPE] first!</span>")
+		return
+
+	if(user.put_in_any_hand_if_possible(SPE))
+		to_chat(user, "<span class='notice'>You remove \the [SPE] from the equipment system.</span>")
+		max_passengers -= SPE.occupant_mod
+		cargo_hold.storage_slots -= SPE.storage_mod["slots"]
+		cargo_hold.max_combined_w_class -= SPE.storage_mod["w_class"]
+		SPE.removed(user)
+		SPE.my_atom = null
+		equipment_system.vars[slot] = null
+	else
+		to_chat(user, "<span class='warning'>You need an open hand to do that.</span>")
 	return
 
+
+/obj/spacepod/hear_talk/hear_talk(mob/M, var/msg)
+	cargo_hold.hear_talk(M, msg)
+	..()
+
+/obj/spacepod/hear_message(mob/M, var/msg)
+	cargo_hold.hear_message(M, msg)
+	..()
+
+/obj/spacepod/proc/return_inv()
+
+	var/list/L = list(  )
+
+	L += src.contents
+
+	for(var/obj/item/weapon/storage/S in src)
+		L += S.return_inv()
+	for(var/obj/item/weapon/gift/G in src)
+		L += G.gift
+		if (istype(G.gift, /obj/item/weapon/storage))
+			L += G.gift:return_inv()
```

Comment:
Gonna fix this instance up, but it's just a copy pasta of the internal inventory calls that are needed for it to work so Someone should probably go clean the colons out of anything with internal storage since you mention it.